Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -822,30 +822,46 @@ ErrorOr<void> VirtualFileSystem::symlink(Credentials const& credentials, StringV
     return {};
 }
 
+// https://pubs.opengroup.org/onlinepubs/9699919799/functions/rmdir.html
 ErrorOr<void> VirtualFileSystem::rmdir(Credentials const& credentials, StringView path, Custody& base)
 {
     RefPtr<Custody> parent_custody;
     auto custody = TRY(resolve_path(credentials, path, base, &parent_custody));
     auto& inode = custody->inode();
 
-    // FIXME: We should return EINVAL if the last component of the path is "."
-    // FIXME: We should return ENOTEMPTY if the last component of the path is ".."
+    auto last_component = KLexicalPath::basename(path);
 
+    // [EINVAL] The path argument contains a last component that is dot.
+    if (last_component == "."sv)
+        return EINVAL;
+
+    // [ENOTDIR] A component of path names an existing file that is neither a directory
+    //           nor a symbolic link to a directory.
     if (!inode.is_directory())
         return ENOTDIR;
 
+    // [EBUSY] The directory to be removed is currently in use by the system or some process
+    //         and the implementation considers this to be an error.
+    // NOTE: If there is no parent, that means we're trying to rmdir the root directory!
     if (!parent_custody)
         return EBUSY;
 
     auto& parent_inode = parent_custody->inode();
     auto parent_metadata = parent_inode.metadata();
 
+    // [EACCES] Search permission is denied on a component of the path prefix,
+    //          or write permission is denied on the parent directory of the directory to be removed.
     if (!parent_metadata.may_write(credentials))
         return EACCES;
 
     if (parent_metadata.is_sticky()) {
-        if (!credentials.is_superuser() && inode.metadata().uid != credentials.euid())
+        // [EACCES] The S_ISVTX flag is set on the directory containing the file referred to by the path argument
+        //          and the process does not satisfy the criteria specified in XBD Directory Protection.
+        if (!credentials.is_superuser()
+            && inode.metadata().uid != credentials.euid()
+            && parent_inode.metadata().uid != credentials.euid()) {
```

Comment:
```suggestion
            && parent_metadata().uid != credentials.euid()) {
```

There is already such a variable so why not use it?